solutions/cf-920E.mdx

```diff
@@ -0,0 +1,232 @@
+---
+id: cf-920E
+source: CF
+title: Connected Components?
+author: Daniel Qiu
+---
+
+
+[Official Editorial](https://codeforces.com/blog/entry/57516)
+
+
+## Solution 1 - Faster DFS
+
+Doing a standard DFS would be too slow. To speed it up, we have to do two
+optimizations.
+
+### Optimization 1
+
+You can store whether there is no edge between two nodes using an array of hashsets, instead of a adjacency list. If the input tells you that nodes $a$ and $b$ do not have an edge between them, you would add $a$ to $\texttt{adj}[b]$, and $b$ to $\texttt{adj}[a]$ 
+This way you can find if there is no edge between two nodes in
+$\mathcal{O}(1)$ time. Using an adjacency matrix would take too much
+memory.
```

Comment:
Something along the lines of "To save memory, we only store non-existent edges in `adj`. To perform a fast lookup of non-existent edges, we can use a HashSet to query this in O(1) time." suffices.

The rest is somewhat redundant.